### PR TITLE
Add a parser for `OpenSCAP` vulnerability/compliance reports in JSON format

### PR DESCRIPTION
**Pull request by akash-manna-sky**
Add a parser for `OpenSCAP` vulnerability/compliance reports in JSON format

See : https://github.com/OpenSCAP/openscap
See : https://www.open-scap.org/resources/documentation/
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests that demonstrate the feature works or the issue is fixed

### CODE REVIEW

**Review by uhafner — APPROVED** [review]

File: src/main/java/edu/hm/hafner/analysis/parser/OpenScapParser.java
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/OpenScapParser.java:1]: Can you please use correct camel case for Java files: `OpenScapParser`

File: src/test/java/edu/hm/hafner/analysis/parser/OpenSCAPParserTest.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+    @Test
+    void shouldNotReportPassResults() {
+        var report = parse("openscap-report.json");
+
+        for (Issue issue : report) {
+            assertThat(issue.getCategory())
+                    .isNotEqualTo("pass")
+                    .isNotEqualTo("notapplicable");
+        }
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/OpenSCAPParserTest.java]: `for` loops are kind of strange in unit tests. Can you replace this one (and the other ones) with:

```
        assertThat(report.get())
                .map(Issue::getCategory)
                .doesNotContain("pass", "notapplicable");

```

File: src/test/java/edu/hm/hafner/analysis/parser/OpenSCAPParserTest.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    void shouldCombineEvidenceAndDescription() {
+        var report = parse("openscap-report-null-handling.json");
+
+        var issue1 = report.get(1);
+        assertThat(issue1.getDescription())
+                .isEqualTo("Detailed description of the issue");
+
+        var issue0 = report.get(0);
+        assertThat(issue0.getDescription())
+                .isEqualTo("Test without rule object");
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/OpenSCAPParserTest.java]: Inline the variables into the assert:


```suggestion
        assertThat(report.get(0)).hasDescription("Test without rule object");
        assertThat(report.get(1)).hasDescription("Detailed description of the issue");
```

File: src/test/java/edu/hm/hafner/analysis/parser/OpenSCAPParserTest.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        assertThat(report.get(0)).hasSeverity(Severity.WARNING_HIGH);
+        assertThat(report.get(1)).hasSeverity(Severity.ERROR);
+        assertThat(report.get(2)).hasSeverity(Severity.WARNING_LOW);
+    }
+
+    @Test
+    void shouldFilterFailAndErrorResults() {
+        var report = parse("openscap-report-null-handling.json");
+
+        for (Issue issue : report) {
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/OpenSCAPParserTest.java]: See above

File: src/test/java/edu/hm/hafner/analysis/parser/OpenSCAPParserTest.java [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+            String category = issue.getCategory();
+            assertThat(category).isIn("fail", "error");
+        }
+    }
+
+    @Test
+    void shouldUseDefaultFileNameWhenMissing() {
+        var report = parse("openscap-report-null-handling.json");
+
+        for (Issue issue : report) {
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/OpenSCAPParserTest.java]: See above

File: src/test/java/edu/hm/hafner/analysis/parser/OpenScapParserTest.java [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+        var report2 = parse("openscap-report-edge-cases.json");
+        assertThat(report2).hasSize(2);
+
+        for (Report r : new Report[]{report1, report2}) {
+            for (Issue issue : r) {
+                assertThat(issue.getCategory()).isIn("fail", "error");
+            }
+        }
+    }
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/OpenScapParserTest.java:173]: Use one file per test.

File: src/test/java/edu/hm/hafner/analysis/parser/OpenSCAPParserTest.java [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+    void shouldParseMultipleFormats() {
+        var report1 = parse("openscap-report.json");
+        assertThat(report1).hasSize(3);
+
+        var report2 = parse("openscap-report-edge-cases.json");
+        assertThat(report2).hasSize(2);
+
+        for (Report r : new Report[]{report1, report2}) {
+            for (Issue issue : r) {
+                assertThat(issue.getCategory()).isIn("fail", "error");
```
Comment (uhafner) [inline src/test/java/edu/hm/hafner/analysis/parser/OpenSCAPParserTest.java]: Maybe it makes sense to extract the assertion from above into a method and reuse it here as well.

**Review by uhafner — CHANGES_REQUESTED** [review]
Sorry, I forgot to review the test cases...

**Review by uhafner — APPROVED** [review]

**Comment by akash-manna-sky** [comment]
Thanks!
